core/kernel/ldelf_syscalls.c

```diff
@@ -51,7 +51,7 @@ TEE_Result ldelf_syscall_map_zi(vaddr_t *va, size_t num_bytes, size_t pad_begin,
 	if (!mobj)
 		return TEE_ERROR_OUT_OF_MEMORY;
 	res = vm_map_pad(uctx, va, num_bytes, prot, vm_flags,
-			 mobj, 0, pad_begin, pad_end, 0);
+			 mobj, 0, pad_begin, pad_end, 0, true);
```

Comment:
`ldelf` is the only user space component that calls `vm_map_pad()` and friends with a user space pointer `vaddr_t *va` in the argument. I'd rather read and write the `va` pointer in `ldelf_syscall_map_zi()` and `ldelf_syscall_map_bin()` as needed.

Reply:
not addressed.
Discard changes in `vm_map_pad()` and `vm_map()` and instead use something like:
```c
	vaddr_t _va = 0;

	(...)

	res = GET_USER(_va, va);
	if (!res)
		res = vm_map_pad(uctx, _va, num_bytes, prot, vm_flags,
				 mobj, 0, pad_begin, pad_end, 0);
```
Ditto in `ldelf_syscall_map_bin()`.